src/LsCommand.cc

```diff
@@ -0,0 +1,186 @@
+/* -*- Mode: C++; tab-width: 8; c-basic-offset: 2; indent-tabs-mode: nil; -*- */
+
+#include <algorithm>
+#include <assert.h>
+#include <vector>
+
+#include "Command.h"
+#include "dirent.h"
+#include "main.h"
+#include "TraceStream.h"
+#include "util.h"
+
+using namespace std;
+
+class LsCommand : public Command {
+public:
+  virtual int run(vector<string>& args);
+
+protected:
+  LsCommand(const char* name, const char* help) : Command(name, help) {}
+
+  static LsCommand singleton;
+};
+
+LsCommand LsCommand::singleton(
+    "ls", " rr ls [OPTION]...\n"
+          "  -l, --long-listing use a long listing format \n "
+          "     (trace name | command line | start time | duration | size)\n"
```

Comment:
If you're going to describe the format, put the components in the same order as you output them. In particular the command line goes last (as it should!).